### PR TITLE
GICv3 emulation preparation PR

### PR DESCRIPTION
**Pull request by kent-mcleod**
These changes are the first few commits from https://github.com/seL4/seL4_projects_libs/pull/58/ which reorganize the vgic code to allow adding support for GICv3.

For reviewing, these changes shouldn't be changing any behavior of the current implementation and are only supposed to be moving code around with a few function prototype changes, eg `static` -> `static inline`.

### CODE REVIEW

File: libsel4vm/src/arch/arm/vgic/vgic_v2.c
Comment (axel-h) [inline libsel4vm/src/arch/arm/vgic/vgic_v2.c:70]:
```suggestion
    return vgic->dist;
```
This seems a merge artifact.

Comment (kent-mcleod) [reply libsel4vm/src/arch/arm/vgic/vgic_v2.c:70]: Thanks for spotting. I moved the priv_get_dist changes into a later commit so that the move commits didn't also modify the code.

File: libsel4vm/src/arch/arm/vgic/virq.h
Comment (axel-h) [inline libsel4vm/src/arch/arm/vgic/virq.h:159]: This function does not really belong here, it would fit better in `vgic_v2.c` even if this duplicates code for GICv3 then. I have a branch where `struct vgic_dist_device` is dropped completely and this is all part of the `struct vgic` then, but have to check how this would work with the GICv3 code then.

Comment (kent-mcleod) [reply libsel4vm/src/arch/arm/vgic/virq.h:159]: What you say makes sense about the function staying in vgic_v2.c, I need to check with the old version whether it had it in there or whether I accidentally pulled it in during a rebase. 

I'm not sure whether dropping vgic_dist_device would make the code cleaner and it'd be better to wait until v3 code is merged before we consider IMO.

File: libsel4vm/src/arch/arm/vgic/vgic_v2.c
Comment (axel-h) [inline libsel4vm/src/arch/arm/vgic/vgic_v2.c:179]: It seem this function remains the same in GICv2 and GICv3. Thus I think it would make sense to keep thios in a common file and avoid code duplication. If there are differences in the future, we can still split this up,

File: libsel4vm/src/arch/arm/vgic/vgic_v2.c
Comment (axel-h) [inline libsel4vm/src/arch/arm/vgic/vgic_v2.c:279]: Same here, this function and alsdo `handle_vgic_maintenance()` are the same in GICv2 and GIV3, so I'd keep this in a common file for now.

Comment (Ivan-Velickovic) [reply libsel4vm/src/arch/arm/vgic/vgic_v2.c:279]: They're slightly different since GICv3 uses group 1 while GICv2 uses group 0.

Comment (Ivan-Velickovic) [reply libsel4vm/src/arch/arm/vgic/vgic_v2.c:279]: (only `handle_vgic_maintenance` differs)

Comment (axel-h) [reply libsel4vm/src/arch/arm/vgic/vgic_v2.c:279]: True, but that could be a constant then. My main point at the moment it trying to have GICv3 support with out renaming the source file and moving code, so the change are easier to maintain.

**Review by pingerino — APPROVED** [review]

**Review by axel-h — APPROVED** [review]
I not too happy with some things in the code, but it the greater goal is adding some GICv3 support finally, then I wont block this.

File: libsel4vm/src/arch/arm/vgic/virq.h
Comment (kent-mcleod) [reply libsel4vm/src/arch/arm/vgic/virq.h]: I added a new commit to address this: https://github.com/seL4/seL4_projects_libs/pull/68/commits/0ec24c1c8c13a73a7441bd9290dbab5a46c71424.  I think it's a bit reactive, but still reasonable to fix it this way.

**Review by pingerino — APPROVED** [review]

**Pull request merged**
